### DISCUSSION TITLE
When solving MIP, I want to set a primal bound and a dual-bound for a model before and during the solving, which interfaces should I call?

### DISCUSSION THREAD

**Discussion by linpeng0105** [ERGO-Code/HiGHS#1983] [Q&A] [answered]
When solving MIP, I could get a primal bound (incumbent) and a dual-bound via a stand-alone heuristic.
I want to set the primal bound and the dual bound for a model before and during the solving, which interfaces should I call?

**Comment by jajhall** [comment, marked as answer]
You can pass a (partial) integer feasible solution to highs that give it an initial primal bound. Use `Highs::passSolution`, which has two versions according to whether you're passing a full set of solution values (as a `HighsSolution` instance, with only the `col_value` vector required) or a "sparse" primal solution (as number of values and pointers to arrays of indices and values). HiGHS will attempt to complete this solution and then use it to obtain a primal bound (if it is feasible).

It is not possible to set the dual bound value
